app/models/order.rb

```diff
@@ -1,14 +1,53 @@
 class Order < ApplicationRecord
   include OwnerField
+  include Discard::Model
   acts_as_tenant(:tenant)
 
-  default_scope { order(:created_at => :desc) }
+  default_scope { kept.order(:created_at => :desc) }
 
-  has_many :order_items
+  has_many :order_items, :dependent => :destroy
 
   before_create :set_defaults
 
+  before_discard :discard_order_items
+  before_undiscard :restore_order_items
+
   def set_defaults
     self.state = "Created"
   end
+
+  private
+
+  CHILD_DISCARD_TIME_LIMIT = 30
+
+  def discard_order_items
+    if order_items.map(&:discard).any? { |result| result == false }
+      order_items.kept.each do |item|
+        Rails.logger.error("OrderItem ID #{item.id} failed to be discarded")
```

Comment:
Should we wrap this in a transaction? Or is `before_undiscard` and `before_discard` already wrapped in a transaction? 

I'm wondering if we should stop the discard altogether if we run into issues removing the `order_item`.

Same question below with `restore_order_items`.

Reply:
That's a good question, I don't think it is. I can't find anything about transactions on the gem github (https://github.com/jhawthorn/discard).

However, since we're in a `before` callback, it should be fine because if there's an error it will raise it and then it can't do anything else. I can add stuff to the specs to ease your mind on this, but I tested locally and it doesn't end up deleting the order if one of the order items fails.